README.md

```diff
@@ -1,5 +1,125 @@
 # Kafka Histo Streamer
 
-Consumes event data from kafka and re-streams it as histogram data
+This is a prototype to test and demonstrate Mantid being capable of live processing event data into histogram data, streaming both via Kafka. Subscribes to a topic with event data and publishes it to another topic as histogram data.
 
+## Requirements
 
+Requires Mantid (currently has to be `nightly`).
+
+Virtualenv Python package recommended and used below (`$ pip install virtualenv`).
+
+Running a local demo additionally requires [Docker](https://www.docker.com/) and [ISIS Nexus Streamer](https://github.com/ScreamingUdder/isis_nexus_streamer_for_mantid/).
+
+Already includes the `hs00` schema, and generated code, from the DMSC [Schema Repository](https://github.com/ess-dmsc/streaming-data-types).
+
+
+## Setup
+
+Ensure [Mantid](https://www.mantidproject.org/Main_Page) is installed.
+
+Clone this repository:
+
+```
+$ git clone https://github.com/DMSC-Instrument-Data/kafka_histo_streamer
+```
+
+
+You will want to edit `run.py` and `postprocess.py` to configure `kafka_server` and `kafka_topic` as needed. If you want to run a local demo, we will revisit what these should be set to below.
+
+## Execution
+
+The Histo Streamer must be executed using `mantidpython`. There are currently no arguments:
+
+```
+user@host:~/kafka_histo_streamer $ mantidpython run.py
+```
+
+## Local Demo
+
+Running a local demo requires a Kafka broker, input event data, and Mantid to visualize the output histogram data.
+
+Recommended way of running a local demo:
+
+![diagram of demo setup](media/histo_streamer_diagram.png)
+
+The Nexus Streamer is started first, to provide a Kafka server and event data. The Histo Streamer is then started to process events into histogram data. Finally, an instance of Mantid is used to consume and visualize the histogram data.
+
+#### Additional Setup
+
+To run a local demo, you will need to [install Docker](https://docs.docker.com/install/) and [Docker Compose](https://docs.docker.com/compose/install/), as well as clone the ISIS Nexus Streamer repository:
```

Comment:
It is not possible/recommended to run this via `docker-machine` i.e Docker should be running natively

Reply:
It should be possible, but not quite as straightforward. I'll add a note.